File: gitdot-api-ts/src/endpoint/question/update-answer.ts
import { z } from "zod";

import { AnswerResource } from "../../resource";

export const UpdateAnswerRequest = z.object({
  body: z.string(),
});
export type UpdateAnswerRequest = z.infer<typeof UpdateAnswerRequest>;

export const UpdateAnswer = {
  path: "/repository/{owner}/{repo}/question/{number}/answer/{answer_id}",
  method: "PATCH",
  request: UpdateAnswerRequest,
  response: AnswerResource,
} as const;
export type UpdateAnswer = typeof UpdateAnswer;

File: gitdot-api/src/endpoint/question/update_answer.rs
use serde::{Deserialize, Serialize};

use crate::{endpoint::Endpoint, resource::question::AnswerResource};

pub struct UpdateAnswer;

impl Endpoint for UpdateAnswer {
    const PATH: &'static str = "/repository/{owner}/{repo}/question/{number}/answer/{answer_id}";
    const METHOD: http::Method = http::Method::PATCH;

    type Request = UpdateAnswerRequest;
    type Response = UpdateAnswerResponse;
}

#[derive(ApiRequest, Debug, Serialize, Deserialize)]
pub struct UpdateAnswerRequest {
    pub body: String,
}

pub type UpdateAnswerResponse = AnswerResource;
